File: benchmarks/workloads/sum_range_xor/python/bench.py
import sys

def range_gen(n):
    i = 0
    while i < n:
        yield i
        i += 1

def main():
    n = int(sys.argv[1])
    a = 0
    for v in range_gen(n):
        a ^= v
    print(f"xor = {a}")

if __name__ == "__main__":
    main()

File: benchmarks/workloads/sum_range_xor/javascript/bench.js
function* rangeGen(n) {
    for (let i = 0n; i < n; i++) yield i;
}

const n = BigInt(process.argv[2]);
let a = 0n;
for (const v of rangeGen(n)) {
    a ^= v;
}
console.log(`xor = ${a}`);
